ZenPacks/zenoss/OpenStackInfrastructure/apiclients/txapiclient.py

```diff
@@ -99,9 +99,22 @@ class NovaClient(BaseClient):
     hypervisorstats = api('/os-hypervisors/statistics')
     hypervisor_detail_id = api('/os-hypervisors/{param[hypervisor_id]}')
     images = api('/images/detail')
-    servers = api('/servers/detail?all_tenants=1')
+    servers = None
     services = api('/os-services')
 
+    # Augment __init__ because of client context dependency
+    def __init__(self, *args, **kwargs):
+        super(NovaClient, self).__init__(*args, **kwargs)
+        self.initialize(self.session_manager.username)
+
+    # Class method to initialize given username context
+    @classmethod
+    def initialize(cls, username):
+        if username == 'admin':
```

Comment:
I don't think the same API methods should ever be making different api calls behind the scenes- i think our api clients should be as thin as possible and just have each method make one api call and return the data.  Especially we don't want magic changes in behavior based on the useername.   (there's nothing to say that the username would even be 'admin' in a particular environment- that's definitely a bad assumption) 

The modeler needs to call the all_tenants API, and only if that fails, it needs to log a warning and fall back to the single user call.  So this logic belongs in the modeler code